gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/PubSub.java

```diff
@@ -0,0 +1,119 @@
+ * limitations under the License.
+ */
+
+package com.google.gcloud.pubsub;
+
+import com.google.gcloud.Page;
+import com.google.gcloud.Service;
+
+import java.util.List;
+import java.util.concurrent.Future;
+
+/**
+ * An interface for Google Cloud Pub/Sub.
+ *
+ * @see <a href="https://cloud.google.com/pubsub/">Google Cloud Pub/Sub</a>
+ */
+public interface PubSub extends Service<PubSubOptions> {
+
+  class ListOption {
+    // page size
+    // page token
+  }
+
+  class PullOption {
+    // bool return_immediately = 2;
+    // int32 max_messages = 3;
+  }
+
+  // topics
+  //////////////////////
+  Topic createTopic(TopicInfo topic);
+
+  Future<Topic> createTopicAsync(String topic);
+
+  // null if not found
+  Topic getTopic(String topic);
+
+  Future<Topic> getTopicAsync(String topic);
+
+  // false if not found
+  boolean deleteTopic(String topic);
+
+  Future<Boolean> deleteTopicAsync(String topic);
+
+  Page<Topic> listTopics(ListOption... options);
+
+  // todo: consider AsyncPage that has nextPageAsync
+  Future<Page<Topic>> listTopicsAsync(ListOption... options);
+
+  String publish(String topic, Message message);
+
+  Future<String> publishAsync(String topic, Message message);
+
+  List<String> publish(String topic, Message first, Message... other);
+
+  Future<List<String>> publishAsync(String topic, Message first, Message... other);
+
+  // subscriptions
+  ////////////////////////////
+  Subscription createSubscription(SubscriptionInfo subscription);
+
+  Future<Subscription> createSubscriptionAsync(SubscriptionInfo subscription);
+
+  // null if not found
+  Subscription getSubscription(String subscription);
+
+  Future<Subscription> getSubscriptionAsync(String topic);
+
+  // false if not found
+  boolean deleteSubscription(String subscription);
+
+  Future<Boolean> deleteSubscriptionAsync(String subscription);
+
+  Page<Subscription> listSubscription(ListOption... options);
+
+  // todo: consider AsyncPage that has nextPageAsync
+  Future<Page<Subscription>> listSubscriptionAsync(ListOption... options);
+
+  Page<Subscription> listSubscription(String topic, ListOption... options);
+
+  // todo: consider AsyncPage that has nextPageAsync
+  Future<Page<Subscription>> listSubscriptionAsync(String topic, ListOption... options);
+
+  // Ack options:
+  // 1) replace return value to ReceivedMessage (Message + functional ack)
+  // 2) like (1) but with no auto-renew, so provide a function for renew
+  // 3) rename Message to MessageInfo and make Message functional with ack
+  // 4) return a "special" list (extends List) but has a way to "ackSoFar"
+  // 5) instead of List use callback and auto-acknowledge per callback (and auto-renew)
+  // ** Auto renew means, using a separate thread.
+  List<Message> pull(String subscription, PullOption... options);
+
+  Future<List<Message>> pullAsync(String subscription, PullOption... options);
+
+
+  // Note regarding Date types:
```

Comment:
Do you mean Date or data here? Just to make sure I understood the following correctly :)